react-select/react-select.d.ts

```diff
@@ -19,48 +19,75 @@ declare namespace ReactSelect {
         clearableValue?: boolean;
     }
 
+    interface MenuRendererProps {
+        /**
+         * The currently focused option; should be visible in the menu by default.
+         * default {}
+         */
+        focusedOption: Option;
+
+        /**
+         * Callback to focus a new option; receives the option as a parameter.
+         */
+        focusOption: (option: Option) => void;
+
+        /**
+         * Option labels are accessible with this string key.
+         */
+        labelKey: string;
+
+        /**
+         * Ordered array of options to render.
+         */
+        options: Option[];
+
+        /**
+         * Callback to select a new option; receives the option as a parameter.
+         */
+        selectValue: (option: Option) => void;
+
+        /**
+         * Array of currently selected options.
+         */
+        valueArray: Option[];
+    }
+
     interface ReactSelectProps extends __React.Props<ReactSelect> {
```

Comment:
For the below, refer to https://github.com/JedWatson/react-select/blob/master/src/Select.js#L29 and its contents